### PR TITLE
Kaitie/sprite lab examples

### PR DESCRIPTION
**Pull request by kobryan0619**
**NOTE:** I am looking for feedback on how to write good PRs, so feel free to add comments in here about what should be added or deleted to make this easier to review. 

After talking to Tess, we decided to both make the windows bigger and keep the wonky scroll feature.  You need to scroll on the white part of the window (not the block space). 

### Before
Small window and no option to scroll. 
image

### After

Slightly larger window and can scroll on the left side of the screen.
chrome-capture-2022-10-22

Added text in the pop-up in the editor that notifies folks that the minimum height is 400px.  Note that they can enter smaller numbers, but what will be displayed will always be 400px.
image

## Links

- [jira ticket](https://codedotorg.atlassian.net/jira/software/projects/TEACH/boards/65?selectedIssue=TEACH-23)

## Testing story

Ran and passed tests for both ExampleTest.jsx and ExampleEditorTest.jsx.

## Other notes 
- Currently, I essentially created a "minimum size" to the window of 400 (our default before was 310).  I didn't modify anything in level builder to indicate this, but thought I would get Bethany's thoughts first.  
- Alternatively, we could just add a bit of cushion to any window size entered in level builder

### CODE REVIEW

File: apps/src/lib/levelbuilder/code-docs-editor/ExampleEditor.jsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const APP_DISPLAY_OPTIONS = {
   codeFromCodeField: 'Display app with code from code field above'
 };
 
-const DEFAULT_EMBED_HEIGHT = 310;
```
Comment (kobryan0619) [inline apps/src/lib/levelbuilder/code-docs-editor/ExampleEditor.jsx:13]: Changed this default to be just a bit larger.

File: apps/src/templates/codeDocs/Example.jsx [hunk 2]
```diff
@@ -36,17 +36,24 @@ export default function Example({example, programmingEnvironmentName}) {
       const embedUrl = example.app.endsWith('embed_app_and_code')
         ? example.app
         : example.app + '/embed_app_and_code';
+      const enteredHeight =
```
Comment (kobryan0619) [inline apps/src/templates/codeDocs/Example.jsx:39]: My thinking around this was that if a curriculum writer chooses a height for the window that is larger than the 310 (the old default) we will use that height for the window.  Anything smaller than that will get made 40% larger operating under the belief that a larger window is better than a smaller one.

File: apps/src/templates/codeDocs/Example.jsx [hunk 3]
```diff
@@ -36,17 +36,24 @@ export default function Example({example, programmingEnvironmentName}) {
...
           <div>
             {content}
-            <iframe
-              src={embedUrl}
-              style={{
-                width: '100%',
-                height: Number(example.embed_app_with_code_height) || 310
-              }}
-            />
+            <div style={{height: enteredHeight, overflow: 'scroll'}}>
```
Comment (kobryan0619) [inline apps/src/templates/codeDocs/Example.jsx:47]: Added the div to force the scroll for the iframe

File: apps/src/templates/codeDocs/Example.jsx [hunk 4]
```diff
@@ -36,17 +36,24 @@ export default function Example({example, programmingEnvironmentName}) {
...
+              <iframe
+                src={embedUrl}
+                style={{
+                  width: '100%',
+                  height: enteredHeight * 1.5,
```
Comment (kobryan0619) [inline apps/src/templates/codeDocs/Example.jsx]: This multiplier means that the user will be able to scroll through 50% more of the code.

File: apps/src/lib/levelbuilder/code-docs-editor/ExampleEditor.jsx [hunk 5]
```diff
@@ -69,7 +69,8 @@ export default function ExampleEditor({example, updateExample}) {
         Example App iframe Embed Height
         <HelpTip>
           The height of the iframe, in pixels, to use when displaying an app
-          with the "Embed app with code" display type
+          with the "Embed app with code" display type. Note that the minimum
+          height is 400px.
```
Comment (kobryan0619) [inline apps/src/lib/levelbuilder/code-docs-editor/ExampleEditor.jsx:73]: Added comment about minimum height

File: apps/src/templates/codeDocs/Example.jsx [hunk 6]
```diff
@@ -36,17 +36,24 @@ export default function Example({example, programmingEnvironmentName}) {
       const embedUrl = example.app.endsWith('embed_app_and_code')
         ? example.app
         : example.app + '/embed_app_and_code';
+      const enteredHeight =
+        Number(example.embed_app_with_code_height) > 400
+          ? Number(example.embed_app_with_code_height)
```
Comment (kobryan0619) [inline apps/src/templates/codeDocs/Example.jsx:41]: Forces the height to be 400 px even if something smaller is entered.

File: apps/src/templates/codeDocs/Example.jsx (hunk 2 above)
Comment (kobryan0619) [inline apps/src/templates/codeDocs/Example.jsx:39]: Forces the height of the frame to be 400px

File: apps/src/templates/codeDocs/Example.jsx (hunk 4 above)
Comment (epeach) [reply apps/src/templates/codeDocs/Example.jsx]: Does this mean there is a 'cap' on how much we can scroll (assuming minimum height (400), we could scroll a max of 600px)?

File: apps/src/templates/codeDocs/Example.jsx [hunk 7]
```diff
@@ -36,17 +36,24 @@ export default function Example({example, programmingEnvironmentName}) {
...
-                height: Number(example.embed_app_with_code_height) || 310
-              }}
-            />
+            <div style={{height: enteredHeight, overflow: 'scroll'}}>
+              <iframe
+                src={embedUrl}
+                style={{
+                  width: '100%',
+                  height: enteredHeight * 1.5,
+                  overflow: 'scroll'
```
Comment (epeach) [inline apps/src/templates/codeDocs/Example.jsx]: Is this scroll needed - or just the one on the surrounding div?

Comment (kobryan0619) [reply apps/src/templates/codeDocs/Example.jsx]: Good point - I just removed it and re-tested - you're right!

File: apps/src/templates/codeDocs/Example.jsx (hunk 4 above)
Comment (kobryan0619) [reply apps/src/templates/codeDocs/Example.jsx]: Yes, that is what it means... I couldn't find a cleaner solution.  We could just make a really big "scroll space" (just set it to 1000px), but that leads to some weird white space at times... I am open to other solutions too!

**Comment by kobryan0619** [comment]
I addressed Erin's comments.

File: apps/src/templates/codeDocs/Example.jsx (hunk 4 above)
Comment (epeach) [reply apps/src/templates/codeDocs/Example.jsx]: I think this is a good solution - just wanted to make sure I fully understood it!

**Review by epeach — APPROVED** [review]
Looks great! :shipit:

**Pull request merged**